hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -807,6 +807,76 @@ def _track_subtype(self, init_values):
                 return self.env.ref("hr_timesheet_sheet.mt_timesheet_approved")
         return super()._track_subtype(init_values)
 
+    # ------------------------------------------------
+    # to prevent sending emails to the reviewer
+    # ------------------------------------------------
+
+    def get_reviewers(self, company, record_id):
+        """Get list of possible reviewers id
+
+        Args:
+            company (record): active timesheet company
+            record_id (int): active timesheet id
+
+        Returns:
+            reviewer_ids (list): list of possible reviewers id
+        """
+        # serach active timesheet
+        sheet = self.env["hr_timesheet.sheet"].search(
+            [("company_id", "=", company.id), ("id", "=", record_id)]
+        )
+        # Create list of possible reviewers
+        reviewer_ids = []
+        reviewer = sheet._get_subscribers()
+        reviewer_ids += [id for id in reviewer.ids if id not in reviewer_ids]
+
+        # Check reviewer exist
+        if (
+            sheet.reviewer_id
+            and sheet.reviewer_id.user_partner_id.id not in reviewer_ids
+        ):
+            reviewer_ids.append(sheet.reviewer_id.user_partner_id.id)
+
+        return reviewer_ids
+
+    def _update_value(self, recipient_data, reviewer_ids):
+        """Remove reviewers from recipient
+
+        Args:
+            recipient_data (dict): values to update
+            reviewer_ids (list): list of reviewers id
+
+        Returns:
+            recipient_data (dict): updated values
+        """
+
+        if recipient_data:
+            # list of recipients without reviewers
+            sanitize_recipient_ids = []
+            for recipient in recipient_data["partners"]:
+                if (
+                    recipient["id"] not in reviewer_ids
+                    and recipient["id"] not in sanitize_recipient_ids
+                ):
+                    sanitize_recipient_ids.append(recipient)
+            recipient_data["partners"] = sanitize_recipient_ids
+
+            return recipient_data
+
+    def _notify_compute_recipients(self, message, msg_vals):
+        model_name = message.model
```

Comment:
Why? is there any chance that message sent from tis model will have another model instead?

Reply:
resolved